### ISSUE TITLE
Hibernate always bump version upon merge calls involving entities without change 

### ISSUE THREAD

**Issue by emmanuelbernard** [quarkusio/quarkus#7770]
Upon merge of a reattached entity deserialized from Jackson and Lombok, we see a different of behavior between 1.2.1.Final and 1.3.1.CR2
See https://github.com/quarkusio/quarkus/issues/7462 for the original info.
Possibly related to https://github.com/quarkusio/quarkus/issues/2815

This issue is here to follow up specifically on the version issue.

**Comment by emmanuelbernard** [comment]
CC @masini 
So Sanne and I did a debug session. We have a version of your version bump test that does not use lombok, and yours that uses lombok.
What we found is that in Quarkus 1.2.1.Final, the lombok entity does not implement `SelfDirtinessTracker` which makes the test in `AbstractEntityEntry.onFlushEntity` might be dirty field to be false because `AbstractEntityEntry.isUnequivocallyNonDirty` does not use the `SelfDirtinessTracker`

Now it turns out that due to this, your version test works as expected with Lombok but that's a bug hiding a bug :) 

Now in Quarkus 1.3.0.CR2, Lombok works as expected and implements `SelfDirtinessTracker`. Likely because we improved the classloader infrastructure. But now we surface this bug on version.
And @Sanne strongly suspects we are back to this issue https://github.com/quarkusio/quarkus/issues/2815
